api/python/provisioner/utils.py

```diff
@@ -315,3 +319,92 @@ def generate_random_secret():
     return ''.join(
         [secrets.choice(seq=passwd_seed) for index in range(passwd_strength)]
     )
+
+
+def load_checksum_from_str(hash_str: str) -> Tuple[bytes, str, str]:
+    """
+    Function helper to load hash checksum from specially constructed string.
+    It parses the hash type, hash sum and file name information from
+    provided string.
+
+    Supported formats of checksum string:
+
+    1. <hash_type>:<check_sum> <file_name>
+    2. <hash_type>:<check_sum>
+    3. <check_sum> <file_name>
+    4. <check_sum>
+
+    where
+    <hash_type> - one of the values from `config.HashType` enumeration
+    <check_sum> - hexadecimal representation of hash checksum
+    <file_name> - a file name to which <hash_type> and <hash_sum> belongs to
+
+    Parameters
+    ----------
+    hash_str: str
+        Specially formatted string with hash data
+
+    Returns
+    -------
+    Tuple
+        return tuple with parsed information:
+            (<check_sum>, <hash_type>, <file_name>)
+        each tuple element can be None
+    """
+    hash_type = None
+    hash_data = None
+    file_name = None
+
+    if hash_str:
+        search = HASH_REGEX.search(hash_str)
```

Comment:
I did such approach in the first version of this PR:

https://github.com/Seagate/cortx-prvsnr/pull/1033/commits/16ec7f99bea06e257869f279789d9a5736984624

But  I don't like it. regex approach is better:
* it takes less code
* do parsing and validation
* easy modify to be tolerant for mistakes in data format, e.g. multiple spaces between hash type and hash sum
* it matches files with `_`:
```python
>>> import re
>>> regex = re.compile(r"(?:([\w\-]+):)?([A-Za-z0-9]+)\s*([\w\-]+\.[A-Za-z0-9]*)?")
>>> print(regex.search("md5:fefa1db67588d2783b83f07f4f5beb5c  single_cortx.iso").groups())
('md5', 'fefa1db67588d2783b83f07f4f5beb5c', 'single_cortx.iso')
```

Reply:
Just to be hornets:
* The first 2 examples are not about the latest version of regex. It works correctly with these cases
* About the last example: it is not correct. Regex correctly parses the hash value. Problem here that `fefa1db67588d2783b83f07f4f5be` has length 29 symbols. It means that `bytes.fromhex` fails because the string is not aligned. The same is true in the case of `split` implementation. It doesn't improve here anything.

I switched implementation to the usage of `split()`.

Reply:
Done

Reply:
I agree with your concerns and actually ok with regex. I just wanted to emphasize that neither regex nor split here is required for validation. Generally we need something (regex or split) just to get space-separated fields and do the validation later. Thanks